cmd/internal/kube/secret.go

```diff
@@ -0,0 +1,127 @@
+	Short: "Generate a Kubernetes secret config for Akita",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		key, secret, err := cmderr.RequireAPICredentials("Akita API key is required for Kubernetes Secret generation")
+		if err != nil {
+			return err
+		}
+
+		err = handleSecretGeneration(namespace, key, secret, output)
+		if err != nil {
+			return err
+		}
+
+		printer.Infoln("Generated Kubernetes secret config to ", output)
+		return nil
+	},
+}
+
+// Represents the input used by secretTemplate
+type secretTemplateInput struct {
+	Namespace string
+	APIKey    string
+	APISecret string
+}
+
+// Generates a Kubernetes secret config file for Akita
+func handleSecretGeneration(namespace, key, secret, output string) error {
+
+	input := secretTemplateInput{
+		Namespace: namespace,
+		APIKey:    base64.StdEncoding.EncodeToString([]byte(key)),
+		APISecret: base64.StdEncoding.EncodeToString([]byte(secret)),
+	}
+
+	secretFile, err := createSecretFile(output)
+	if err != nil {
+		return cmderr.AkitaErr{Err: errors.Wrap(err, "failed to create output file")}
+	}
+
+	defer secretFile.Close()
+
+	err = secretTemplate.Execute(secretFile, input)
+	if err != nil {
+		return cmderr.AkitaErr{Err: errors.Wrap(err, "failed to generate template")}
+	}
+
+	return nil
+}
+
+// Creates a file at the give path to be used for storing of the generated Secret config
+// If any child dicrectories do not exist, it will be created.
+func createSecretFile(path string) (*os.File, error) {
+	// Split the outut flag value into directory and filename
+	outputDir, outputName := filepath.Split(path)
+
+	// Get the absolute path of the output directory
+	absOutputDir, err := filepath.Abs(outputDir)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to resolve the absolute path of the output directory")
+	}
+
+	// Check if the output file already exists
+	outputFilePath := filepath.Join(absOutputDir, outputName)
+	if _, statErr := os.Stat(outputFilePath); statErr == nil {
+		return nil, errors.Errorf("output file %s already exists", outputFilePath)
+	}
+
+	// Create the output directory if it doesn't exist
+	err = os.MkdirAll(absOutputDir, 0755)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to create the output directory")
+	}
+
+	// Create the output file in the output directory
+	outputFile, err := os.Create(outputFilePath)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create the output file")
+	}
+
+	return outputFile, nil
+}
+
+func init() {
+	var err error
+
+	secretTemplate, err = template.ParseFS(templateFS, "template/akita-secret.tmpl")
+	if err != nil {
+		log.Fatalf("unable to parse kube secret template: %v", err)
+	}
+
+	secretCmd.Flags().StringVarP(
+		&namespace,
+		"namespace",
+		"n",
+		"",
+		"The Kuberenetes namespace the secret should be applied to",
+	)
+	_ = secretCmd.MarkFlagRequired("namespace")
+
+	secretCmd.Flags().StringVarP(&output, "output", "o", "akita-secret.yml", "File to output the generated secret.")
```

Comment:
This should have come up in the design review -- sorry.  I think the correct default, which is most idiomatic to Kubernetes tools, is to print to standard output.

The idiomatic usage we are aiming for is something like "akita kube secret | kubectl apply -f -"